trl/trainer/dpo_trainer.py

```diff
@@ -128,13 +130,23 @@ def __init__(
                 )
                 max_prompt_length = 128
 
+            if max_target_length is None & is_encoder_decoder:
```

Comment:
```suggestion
            if max_target_length is None and is_encoder_decoder:
```
+ you can use `model.is_encoder_decoder` if `model` is passed.